File: RandomNPCGenerator/src/Dice.java
import java.util.Random;

/**
 * This rolls a n sided die where n is provided
 * @Author Brady Klein
 * @Date 01/11/18
 */

public class Dice {
    //attributes
    private int sides;
    private Random rand = new Random();
    //constructor
    public Dice(int sides){
        this.sides = sides;
    }
    //method
    public int roll(){
        return rand.nextInt(sides);
    }

    public int getStat(){
        int possibles[] = new int[4];
        possibles[0] = rand.nextInt(6);
    }
}

File: Dice.py
import random
"""
 * This rolls a n sided die where n is provided
 * Now implemented in python
 * @Author Brady Klein
 * @Date 01/11/18
 """

class Dice :
    sides = 0

    def __init__(self, side_number):
        self.sides = side_number

    #method
    def roll(self):
        return random.randrange(0, self.sides)

    def get_stat(self):
        possibles = [0, 0, 0, 0]
        possibles[0] = random.randrange(0, 6);
        possibles[1] = random.randrange(0, 6);
        possibles[2] = random.randrange(0, 6);
        possibles[3] = random.randrange(0, 6);
        min_index = 0
        for i in possibles:
            if i < possibles[min_index]:
                min_index = i
        stat = 0
        for i in possibles:
            if i != possibles[min_index]:
                stat += i + 1
        return stat
